File: data/data.code.lua.server/AckChatSendOk.lua
module("AckChatSendOk", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckChatSendOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.channel = 0
	self.uid = 0
	self.uname = ""
	self.content = ""
end


function encode(self)
	local pack = packet.new()
	pack:writeU8(self.channel)
	pack:writeU32(self.uid)
	pack:writeString(self.uname)
	pack:writeString(self.content)
	return pack:encode(Msg.P_ACK_CHAT_SEND_OK)
end

File: examples/golang/server/proto/ack_chat_send_ok.go
package proto

import (
	"packet"
)

type AckChatSendOk struct {
	Channel                  uint8
	Uid                      uint32
	Uname                    string
	Content                  string
}

func (this *AckChatSendOk) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint8(this.Channel)
	pack.WriteUint32(this.Uid)
	pack.WriteString(this.Uname)
	pack.WriteString(this.Content)

	return pack.Encode(P_ACK_CHAT_SEND_OK)
}
